src/main/java/io/kestra/plugin/azure/eventhubs/service/consumer/EventHubConsumerService.java

```diff
@@ -0,0 +1,241 @@
+            .processEventBatch(context -> {
+                PartitionContext partitionContext = context.getPartitionContext();
+                if (!partitions.remove(partitionContext.getPartitionId())) {
+                    if (logger.isTraceEnabled()) {
+                        logger.trace(
+                            "PartitionId={} has already been consumed once. Rejecting events.",
+                            partitionContext.getPartitionId()
+                        );
+                    }
+                    return;
+                }
+                List<EventData> events = context.getEvents();
+
+                // Convert eventData, and invoke listener.
+                converter.convertFromEventData(events).forEach(event -> listener.onEvent(event, partitionContext));
+
+                EventHubNamePartition key = new EventHubNamePartition(
+                    partitionContext.getEventHubName(),
+                    partitionContext.getPartitionId()
+                );
+
+                // Keep checkpoint of the last event in the batch
+                createCheckpoint(context)
+                    .ifPresent(checkpoint -> checkpointsByPartitions.put(key, checkpoint));
+
+                // Increment event counter for the current partition.
+                eventsByEventHubNamePartition
+                    .computeIfAbsent(key, ignored -> new AtomicInteger(0))
+                    .addAndGet(events.size());
+
+                // Check whether all partitions were polled at-least once.
+                if (partitions.isEmpty()) {
+                    // Proactively stop consuming.
+                    latch.countDown();
+                }
+
+            }, consumerContext.maxPollEvents(), consumerContext.maxBatchPartitionWait())
+            // Handle errors
+            .processError(errorContext -> {
+                PartitionContext partitionContext = errorContext.getPartitionContext();
+                logger.error("Failed to process eventHub: {}, partitionId: {} with consumerGroup: {}",
+                    partitionContext.getEventHubName(),
+                    partitionContext.getPartitionId(),
+                    partitionContext.getConsumerGroup(),
+                    errorContext.getThrowable());
+                latch.countDown(); // stop processing immediately.
+            })
+            .buildEventProcessorClient();
+
+        try {
+            listener.onStart();
+            client.start();
+            if (!latch.await(consumerContext.maxDuration().toMillis(), TimeUnit.MILLISECONDS)) {
+                logger.debug("Reached `maxDuration`({}ms) before receiving events EventHub {}.",
+                    consumerContext.maxDuration().toMillis(),
+                    config.eventHubName());
+            }
+        } catch (InterruptedException e) {
+            logger.warn("Task was interrupted before processing EventHub partitions.");
+            Thread.currentThread().interrupt();
```

Comment:
I don't think it's a good idea to catch InterruptedException, as we interrupt thread to kill a task we should not handle it.